Debug App/Sources/Utilities/MetadataParser.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  MetadataParser.swift
+//  Debug App
+//
+//  Created by Niall Quinn on 11/03/24.
+//  Copyright © 2024 Primer API Ltd. All rights reserved.
+//
+
+import Foundation
+
+
+struct MetadataParser {
+
+    func parse(_ metadata: String?) -> [String: Any] {
+        guard let metadata = metadata else { return [:] }
+
+        if let jsonData = metadata.data(using: .utf8),
+           let jsonObject = try? JSONSerialization.jsonObject(with: jsonData, options: []) as? [String: Any] {
+            return jsonObject
+        }
+
+        let keyValuePairs = metadata
```

Comment:
Its not a query string we are dealing with, its a text blob of key value pairs each on their own lines. Ill see how it copes with the \n character, but might simplify the key value pair decoding in isolation. 

Example of metadata (it can also be JSON, but that is caught earlier):
```
scenario=KLARNA
someOtherKey=VALUE
```